bin/live/pycbc_live_collated_dq_trigger_rates

```diff
@@ -0,0 +1,157 @@
+ar_flag_name = args.analysis_flag_name.format(ifo=args.ifo)
+day_seg = Segment(args.gps_start_time, args.gps_end_time)
+observing_flag = DataQualityFlag.query(ar_flag_name, day_seg)
+observing_segs = observing_flag.active
+livetime = observing_flag.livetime
+logging.info(f'Found {livetime} seconds of observing time at {args.ifo}.')
+
+# for each segment, check how much time was dq flagged
+flagged_time = 0
+frame_type = args.analysis_frame_type.format(ifo=args.ifo)
+dq_channel = args.dq_channel.format(ifo=args.ifo)
+dq_ok_channel = args.dq_ok_channel.format(ifo=args.ifo)
+for seg in observing_segs:
+    frames = gwdatafind.find_urls(
+        args.ifo[0],
+        frame_type,
+        seg[0],
+        seg[1],
+    )
+    tsdict = TimeSeriesDict.read(
+        frames,
+        channels=[dq_channel, dq_ok_channel],
+        start=seg[0],
+        end=seg[1],
+    )
+    dq_ok_flag = (tsdict[dq_ok_channel] == 1).to_dqflag()
+    dq_flag = (tsdict[dq_channel] <= args.dq_thresh).to_dqflag()
+    flagged_time += (dq_flag & dq_ok_flag).livetime
+logging.info(f'Found {flagged_time} seconds of dq flagged time at {args.ifo}.')
+
+bg_livetime = livetime - flagged_time
+state_time = np.array([bg_livetime, flagged_time])
+
+# read in template bins
+logging.info(f'Reading template bins from {args.template_bin_file}')
+template_bins = {}
+num_templates = 0
+with HFile(args.template_bin_file, 'r') as bin_file:
+    f_lower = bin_file.attrs['f_lower']
+    bank_file = bin_file.attrs['bank_file']
+    bin_string = bin_file.attrs['background_bins']
+
+    # only ever one group in this file
+    grp = bin_file[list(bin_file.keys())[0]]
+    num_bins = len(grp.keys())
+    for k in grp.keys():
+        template_bins[k] = grp[k]['tids'][:]
+        num_templates += len(template_bins[k])
+
+# for each bin, get total number of triggers and number of dq flagged triggers
+bin_total_triggers = np.zeros(num_bins)
+bin_dq_triggers = np.zeros(num_bins)
+
+logging.info(f'Reading triggers from {args.trigger_file}')
+with HFile(args.trigger_file, 'r') as trigf:
+    for bin_name in template_bins.keys():
+        # bins are named as 'bin{bin_num}'
+        bin_num = int(bin_name[3:])
+        template_ids = template_bins[bin_name]
+        for template_id in template_ids:
+            # get dq states for all triggers with this template
+            # dq state is either 0 or 1 for each trigger
+            dq_key = trigf[f'{args.ifo}/dq_state_template'][template_id]
+            dq_states = trigf[f'{args.ifo}/dq_state'][dq_key]
+
+            # update trigger counts
+            bin_total_triggers[bin_num] += len(dq_states)
+            bin_dq_triggers[bin_num] += np.sum(dq_states)
+
+# write outputs to file
+logging.info(f'Writing results to {args.output}')
+with h5.File(args.output, 'w') as f:
+    ifo_group = f.create_group(args.ifo)
+    ifo_group.create_dataset('observing_livetime', data=livetime)
+    ifo_group.create_dataset('dq_flag_livetime', data=flagged_time)
```

Comment:
I've found the issue with using HFile, [this line](https://github.com/gwastro/pycbc/blob/93f2856ae1b2c88d5f19a0d87ac12db5a234eb66/pycbc/io/hdf.py#L52) uses hasattr('dtytpe') to check it is is a numpy array, which is needed to allow the fletcher32 keyword to be set.

However numpy number-types have dtype attributes as well. The options are either:
 - change that line to use `isinstance(data, np.ndarray)` instead [should work]
 - change this line to give dq_flag_livetime as an attribute [will work]
 - Change this line to cast dq_flag_livetime as a numpy array before saving it [I think this will work, and allow this dataset to be checksummed]

Reply:
I think changing that line to use `isinstance(data, np.ndarray)` is the cleanest fix. It will also prevent this issue from arising again in the future for some other unsuspecting student.